htdocs/stripe/class/stripe.class.php

```diff
@@ -90,15 +90,12 @@ public function getStripeAccount($mode = 'StripeTest', $fk_soc = 0)
 		global $conf;
 
 		$sql = "SELECT tokenstring";
-		$sql .= " FROM ".MAIN_DB_PREFIX."oauth_token";
-		$sql .= " WHERE entity = ".$conf->entity;
-		$sql .= " AND service = '".$mode."'";
+		$sql .= " FROM ".MAIN_DB_PREFIX."oauth_token WHERE";
+		if (empty($fk_soc)) $sql .= " entity = ".$conf->entity. " AND";
+		$sql .= " service = '".$mode."'";
 		if ($fk_soc > 0) {
 			$sql .= " AND fk_soc = ".$fk_soc;
 		}
-		else {
```

Comment:
Are you sure, you must remove this.
It means if fk_soc is not provided, and there is several record for different companies, we may get the one of a company randomly when we want the global one.